src/main/kotlin/com/aemtools/completion/htl/completionprovider/FileVariablesResolver.kt

```diff
@@ -127,11 +128,9 @@ object FileVariablesResolver {
                     startsWith(DATA_SLY_LIST) -> {
                         val (itemName, itemListName) = extractItemAndItemListNames(this)
 
-                        val itemClass = resolveClassForDataSlyList(it)
-
                         val tag = it.findParentByType(XmlTag::class.java) ?: return result
 
-                        if (position.within(tag)) {
+                        if (position.isWithin(tag)) {
```

Comment:
I've reverted the "wrong" refactoring changes, thanks,
But I'm not quite sure if I understand the second question. What do you mean under "define requirements for documentation"?
If you mean the code documentation, my rule was - if the code isn't self-evident -> explain it via docs
Isn't it sufficient? Or you want some general explanation of the project?